datahub-actions/src/datahub_actions/plugin/source/kafka/kafka_event_source.py

```diff
@@ -201,16 +217,69 @@ def close(self) -> None:
             self.running = False
             self.consumer.close()
 
-    def ack(self, event: EventEnvelope) -> None:
-        self.consumer.commit(
+    def _with_retry(self, func: Callable, *args, **kwargs) -> Any:  # type: ignore
+        curr_attempt = 0
+        max_attempts = self.source_config.commit_retry_count
+        backoff = 0.5
+
+        while curr_attempt < max_attempts:
+            try:
+                return func(*args, **kwargs)
+            except Exception as e:
+                logger.error(str(e))
+
+                curr_attempt = curr_attempt + 1
+                if curr_attempt >= max_attempts:
+                    return
```

Comment:
minor: any log.warn we'd like to write out for viz here? or will it be caught and logged at the parent?

Reply:
added warning